### DISCUSSION TITLE
Can you give me an example?(I need help!)

### DISCUSSION THREAD

**Discussion by ileonemil** [ramnes/notion-sdk-py#56] [Q&A] [answered]
Hi! I'm new to coding...can you please give me some examples of anything? Especially how to create a block...Python continuously gives me an error KeyError: 'secret_...' but I'm sure that all the codes are ok. This is what I was searching to do:

```python
from pprint import pprint
from notion_client import Client
import os

notion = Client(auth=os.environ["secret_123456789012345678901234567890"])

notion.pages.create({
    parent: {
      database_id: '12345678901234567890',
    },
    properties: {
      Name: {
        title: [
          {
            text: {
              content: 'Giargiana',
            }
            }
          ]
        }
      }
    }
)
```

_Edit by ramnes: I removed a token and beautified code._

**Comment by aahnik** [comment, marked as answer]
This is a duplicate question. Read this https://github.com/ramnes/notion-sdk-py/discussions/23. Please search existing questions before asking one. You could have typed "KeyError" in the search bar of GitHub discussions. [Learn markdown](https://www.markdownguide.org/extended-syntax/#fenced-code-blocks) to format your code in your question.

------
Read the [Quick Start](https://ramnes.github.io/notion-sdk-py/quick_start/) guide. This guide gives the correct code to try out this library.

You can see [first project example](https://github.com/ramnes/notion-sdk-py/tree/main/examples/first_project) after reading the quick start guide.

Please explore the documentation and readme before asking questions. Also, make sure to google search your error message.

-----------

If you have [google searched "python key error"](https://www.google.com/search?q=python+key+error) you would have got some idea about the mistake you were making.

From your code it seems that you don't know about `os.environ`. Read [this article](https://www.geeksforgeeks.org/python-os-environ-object/) to learn 

-------

> notion = Client(auth=os.environ["secret_123456789012345678901234567890"])

this line is wrong. This is not how you load environment variables.

you must give the name of the environment variable. The key of the `os.environ` mapping is the name of the environment variable.

The correct code is this

```python
notion = Client(auth=os.environ["NOTION_TOKEN"])
```

You must set the value of `NOTION_TOKEN` environment variable.

In your terminal, you can run.

```shell
export NOTION_TOKEN=secret123abc...
```

You can also use a `.env` file and the [`python-dotenv`](https://github.com/theskumar/python-dotenv) library. I have used the python-dotenv library in the [first_project example](https://github.com/ramnes/notion-sdk-py/blob/main/examples/first_project/script.py) and also here https://github.com/ramnes/notion-sdk-py/discussions/23#discussioncomment-743525

**Reply by aahnik** [reply]
hi @ileonemil mark the above as answer, if it helped.

**Reply by ileonemil** [reply]
Hi, thank you for everything. Asap I'll read and do all you've just said, I'm from mac so I have to understand how to create a virtual environment. I'll do that tomorrow, and sorry for duplicating a request, I'm new even to GitHub ahahhah. I'll let you know, thanks again!!

**Reply by ileonemil** [reply]
@aahnik can you please send me an example of a code that creates a block?
also, creating a virtual env i had this: 
```
leo@MacBook-Pro-di-Matteo learn-notion-sdk-py % python -m venv .venv && source .venv/bin/activate
/System/Library/Frameworks/Python.framework/Versions/2.7/Resources/Python.app/Contents/MacOS/Python: No module named venv
leo@MacBook-Pro-di-Matteo learn-notion-sdk-py % 
```

**Reply by aahnik** [reply]
This video by Corey Schafer may help

https://youtu.be/Kg1Yvry_Ydk

Explains creating and using virtual environments in Linux and Mac.

For creating a block call the append method of blocks endpoint. See the reference https://ramnes.github.io/notion-sdk-py/reference/api_endpoints

Until high level pythonic access is available for this library, you need to do the hardwork of writing the block object as a python dictionary.

Read Notion's API reference for the blocks endpoint

https://developers.notion.com/reference/block

**Reply by ileonemil** [reply]
Hi! Thanks for all the resources, they've helped a lot. I finally managed to do what I wanted, but I think I bypassed the virtual env, is that a problem? Anyway now I have to automate the code to run every day, I think I'll use Cron. 
Massive thanks!!!

**Reply by aahnik** [reply]
> I bypassed the virtual env, is that a problem?

That can be a problem in the future. Watch this to learn https://youtu.be/Kg1Yvry_Ydk

**Reply by ileonemil** [reply]
Watched it, it's just that I don't know how to make python running in the venv automatically from the script. I don't want to switch the directory from the terminal every time...

**Reply by aahnik** [reply]
you need to activate the virtual environment to use it. once done, deactivate it. this is helpful, when you have many python projects in your computer. 

the dependencies of different projects will not collide if you use separate virtual environments.

**Reply by ileonemil** [reply]
Ok thank you, you're very kind :)
